addons/stock/models/stock_picking.py

```diff
@@ -523,6 +521,9 @@ def _prepare_pack_ops(self, quants, forced_qties):
         if len(picking_moves.mapped('location_dest_id')) > 1:
             raise UserError(_('The destination location must be the same for all the moves of the picking.'))
```

Comment:
This is unfortunate, because it does not alow to buy products where different push rules do apply, for example raw material and finished goods in one single PO, but on the other hand, it makes the fix herein easier and almost a no brainer.